### ISSUE TITLE
[Bug] Error when generating dbt docs

### ISSUE THREAD

**Issue by mrprigun** [markporoshin/dbt-greenplum#11]
I'm trying to generate documentation for the dbt project that works with Greenplum database. The documentation generates but it return non zero exit code, please take a look at the error bellow.

``` bash
> dbt docs generate       
> Running with dbt=1.2.0
>  Found 40 models, 55 tests, 0 snapshots, 0 analyses, 839 macros, 1 operation, 7 seed files, 0 sources, 0 exposures, 0 metrics
>  
>  Concurrency: 1 threads (target='dev')
>  
>  Done.
>  Building catalog
>  Encountered an error while generating catalog: Runtime Error
  maximum recursion depth exceeded while calling a Python object
>  dbt encountered 1 failure while writing the catalog
> 16:24:36  Catalog written to ......
```

I tried to change driver from greenplum to postgres in my dbt project and it works well without `maximum recursion depth exceeded while calling a Python object` error. Looks like there is an issue somewhere here `dbt/include/greenplum/macros/catalog.sql`.

**Comment by markporoshin** [comment]
Hi, I'm going to look on this problem, but in greenplum adapter we do nothing with documentation. It looks strange

**Comment by markporoshin** [comment]
Do you use partition functionality?

**Comment by mrprigun** [comment]
> Hi, I'm going to look on this problem, but in greenplum adapter we do nothing with documentation. It looks strange

Yep this adapter doesn't do anything with documentation. But if I'll run command `dbt docs generate` dbt will use this adapter to get db schema and types details. Looks like the adapter is not able to proceed that, but postgres adapter can.

> Do you use partition functionality?

No

**Comment by markporoshin** [comment]
I have tested `dbt docs generate` and have no errors. Can you provide simple project to reproduce error?
